lib/ecto/changeset.ex

```diff
@@ -1105,8 +1102,8 @@ defmodule Ecto.Changeset do
         {Map.put(changes, key, change), errors, valid? and relation_valid?}
       :ignore ->
         {changes, errors, valid?}
-      :error ->
-        error = {key, {"is invalid", [type: expected_relation_type(relation)]}}
+      {:error, reason} ->
```

Comment:
nitpick: reason is usually an atom/string/etc, in this case it's `{message, metadata_tuple}`. It's not ideal because we're using word error a lot here, but maybe the variable should be called `error` still (which matches `t:Ecto.Changeset.error/0` type)?

btw, thanks for this PR <3

Reply:
Thanks for pointing it out! It's fixed.